src/core/lib/security/credentials/external/pluggable_auth_external_account_credentials.cc

```diff
@@ -91,13 +134,102 @@ PluggableAuthExternalAccountCredentials::
     }
     output_file_path_ = executable_it->second.string();
   }
+
+  audience_ = options.audience;
+  subject_token_type_ = options.subject_token_type;
+  impersonation_url_ = options.service_account_impersonation_url;
 }
 
 void PluggableAuthExternalAccountCredentials::RetrieveSubjectToken(
-    HTTPRequestContext* ctx, const Options& /*options*/,
+    HTTPRequestContext* /*ctx*/, const Options& /*options*/,
     std::function<void(std::string, grpc_error_handle)> cb) {
   cb_ = cb;
-  // TODO: Execute command
+  if (output_file_path_ != "") {
+    std::string output_file_content = read_file_contents(output_file_path_);
+    if (output_file_content != "") {
+      OnRetrieveSubjectToken(output_file_content);
+      return;
+    }
+  }
+
+  std::vector<std::string> envp = {
+      absl::StrFormat("GOOGLE_EXTERNAL_ACCOUNT_AUDIENCE=%s", audience_),
+      absl::StrFormat("GOOGLE_EXTERNAL_ACCOUNT_TOKEN_TYPE=%s",
+                      subject_token_type_),
+      absl::StrFormat("GOOGLE_EXTERNAL_ACCOUNT_INTERACTIVE=%d", 0),
+      absl::StrFormat("GOOGLE_EXTERNAL_ACCOUNT_IMPERSONATED_EMAIL=%s",
+                      get_impersonated_email(impersonation_url_)),
+      absl::StrFormat("GOOGLE_EXTERNAL_ACCOUNT_OUTPUT_FILE=%s",
+                      output_file_path_)};
+  Subprocess* subprocess = new Subprocess();
+  std::packaged_task<bool(Subprocess*, std::string, std::vector<std::string>,
+                          std::string*, std::string*)>
+      run_executable_task(run_executable);
+  std::future<bool> executable_output_future = run_executable_task.get_future();
```

Comment:
Will this all work with windows without having to change anything assuming the subprocess.communicate and subprocess.start are implemented for it?

Reply:
Most of the stuff in std:: should have implementations in both windows and linux. So hopefully this should work. But the file read needs windows specific code.

Reply:
Oh I think code to read the file already exists, https://github.com/grpc/grpc/blob/2c81c5619e80d69c96496098879620ed973d7b2c/src/core/lib/security/credentials/external/file_external_account_credentials.cc#L110C1-L111C62

from the file based credentials

Reply:
Sweet, I'll use this.